cleanlab_studio/studio/enrichment.py

```diff
@@ -92,3 +108,226 @@ def to_dict(self) -> Dict[str, Any]:
+            **Example 3:** ``regex = (' Explanation:.*', '') will remove everything after and including the words "Explanation:".
+            For instance, the response "True. Explanation: 3+4=7, and 7 is an odd number." would return "True." after the regex replacement.
+        tlm_options (TLMOptions, optional): Options for the Trustworthy Language Model (TLM) to use for data enrichment.
+    """
+
+    prompt: str
+    constrain_outputs: Optional[List[str]]
+    optimize_prompt: Optional[bool]
+    quality_preset: Optional[TLMQualityPreset]
+    regex: Optional[Union[str, Replacement, List[Replacement]]]
+    tlm_options: Optional[TLMOptions]
+
+
+def _validate_enrichment_options(options: EnrichmentOptions) -> None:
+    """Validate the enrichment options."""
+    # Validate the prompt
+    if len(options["prompt"]) == 0:
+        raise ValueError("The 'prompt' parameter must be a non-empty string.")
+    prompt_pattern = r"\$\{.*?\}"
+    if re.search(prompt_pattern, options["prompt"]):
+        raise ValueError(
+            "The 'prompt' parameter should contains at least one dataset column name as '$\{my_column_name\}'."
+        )
+
+    # Validate the regex
+    def _validate_tuple_is_replacement(t: Tuple) -> None:
+        if isinstance(t, tuple) and len(t) == 2 and all(isinstance(x, str) for x in t):
+            return None
+        raise ValueError(REGEX_PARAMETER_ERROR_MESSAGE)
+
+    if "regex" in options:
+        user_input_regex = options["regex"]
+        if user_input_regex:
+            if isinstance(user_input_regex, str):
+                return None
+            elif isinstance(user_input_regex, tuple):
+                _validate_tuple_is_replacement(user_input_regex)
+            elif isinstance(user_input_regex, list):
+                for replacement in user_input_regex:
+                    _validate_tuple_is_replacement(replacement)
+            else:
+                raise ValueError(REGEX_PARAMETER_ERROR_MESSAGE)
+
+
+class EnrichmentResult:
+    """Enrichment result."""
+
+    def __init__(self, results: pd.DataFrame):
+        self._results = results
+
+    @classmethod
+    def from_dict(cls, json_dict: JSONDict) -> EnrichmentResult:
+        raise NotImplementedError()
+
+    def to_list(self) -> List[Tuple[Optional[str], float]]:
+        raise NotImplementedError()
+
+    def details(self) -> pd.DataFrame:
+        return self._results
+
+    def join(self, original_data: pd.DataFrame, *, with_details: bool = False) -> pd.DataFrame:
+        df = self._results
+        joined_data = original_data.join(df, how="left")
+        return joined_data
+
+
+class EnrichmentPreviewResult(EnrichmentResult):
+    """Enrichment preview result."""
+
+    _is_timeout: bool
+    _failed_jobs_count: int
+    _completed_jobs_count: int
+    _final_result_name: str
+
+    @classmethod
+    def from_dict(cls, json_dict: Dict) -> EnrichmentPreviewResult:
+        new_column_name_mapping = json_dict["new_column_name_mapping"]
```

Comment:
I believe we changed this to `enrichment_column_name_mapping`

Reply:
done